ktor-utils/jvm/src/io/ktor/util/collections/ConcurrentMapJvm.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.util.collections
+
+import java.util.*
+
+/**
+ * Ktor concurrent map implementation. Please do not use it.
+ */
+public actual class ConcurrentMap<Key, Value> actual constructor(initialCapacity: Int) : MutableMap<Key, Value> {
+    private val delegate = Collections.synchronizedMap(LinkedHashMap<Key, Value>(initialCapacity))
```

Comment:
Why not use `ConcurrentHashMap` under the hood?

And overall, why expect class instead of just expect factory function like `concurrentMap`?

Reply:
Could you tell me what is the purpose of using ConcurrentHashMap here? The workload is not intensively concurrent.

We need to add `computeIfAbsent` which is missing in common, so we need a dedicated type for that

Reply:
If performance is not the concern for this map, than it's ok. May be better then, to rename it to `SynchronizedMap` instead of `ConcurrentMap` to better reflect, what are the implementation details under the hood?

BTW, looking on code, seems that `Native` implementation can be just moved to common and used on all platforms, just too reduce amount of code.